Documentation/contributing/development/introducing_new_crds.rst

```diff
@@ -0,0 +1,400 @@
+Introducing new CRDs
+====================
+
+Cilium uses a combination of code generation tools to facilitate adding
+CRDs to the Kubernetes instance it is installed on.
+
+These CRDs make themselves available in the generated Kubernetes client
+Cilium uses.
+
+Defining And Generating CRDs
+----------------------------
+
+Paths
+
+::
+
+   pkg/k8s/apis/cilium.io/v2/
+   pkg/k8s/apis/cilium.io/v2alpha1/
+
+Currently, two API versions exist ``v2`` and ``v2alpha1``.
+
+CRDs are defined via Golang structures, annotated with ``marks``, and
+generated with Cilium make file targets.
+
+Marks
+~~~~~
+
+Marks are used to tell ``controller-gen`` *how* to generate the CRD.
+This includes defining the CRD’s various names (Singular, plural,
+group), it’s Scope (Cluster, Namespaced), Shortnames, etc…
```

Comment:
Please let's avoid curly quotes in the doc (I saw single and double ones).

Reply:
Sorry, for this line I meant the apostrophe (`’`). I've also noticed curly quotes elsewhere (`“`/`”`). Probably pandoc trying to be smart.